packages/angular/build/src/utils/index-file/auto-csp.ts

```diff
@@ -288,7 +301,7 @@ function createLoaderScript(srcList: SrcScriptTag[], enableTrustedTypes = false)
     s.type = scriptUrl[1];
     s.async = !!scriptUrl[2];
     s.defer = !!scriptUrl[3];
-    document.body.appendChild(s);
+    document.${inBody ? 'body' : 'head'}.appendChild(s);
```

Comment:
Consider: What if we updated this to support either case like:

```typescript
(document.body ?? document.head).appendChild(s);
```

This way we don't need to track which element we're in? That's probably a slight bundle size regression, but I feel like this should be insignificant enough to not matter? There's probably bigger opportunities here if we wanted to optimize anyways.

For that matter, does it make difference to insert in the head or the body as long as that element exists? Could we just always insert tags in to `head`, even if they originated from the body? Would that cause problems?

Reply:
```
That's probably a slight bundle size regression, but I feel like this should be insignificant enough to not matter?
```

The nit is that the extra characters for ` ?? document.head)` will be included in every loader script, which can scale to the number of `<script src="..."></script>` nodes in cases like:

```
<script src="..."></script>
<script>console.log("hi");</script>
<script src="..."></script>
<script>console.log("hi");</script>
<script src="..."></script>
...
```

since there is one loader script per contiguous block of `<script>` tags. Depending on how many `<script src="..."></script>` tags there are, maybe it isn't a big deal.

```
For that matter, does it make difference to insert in the head or the body as long as that element exists? Could we just always insert tags in to head, even if they originated from the body?
```

For this question, I think (as we found out with the execution order for nodes added through `document.appendChild(...)`) the execution order depends on the time at which the node was inserted into the document rather than where it was inserted into the document-- but one advantage of correctly appending to the appropriate `head` or `body` is that when the Chrome Dev Tools is used to look at the final markup, the loader scripts insert `<script src=...>` tags into where they *would've* been in the DOM, which be a small benefit aesthetically. Do you think the added complexity of adding the scripts to where they would've been in the DOM is worth it for this aesthetic benefit? Thanks!

Reply:
As discussed, we will stop tracking `inBody` and instead go with `document.lastElementChild` to make it even more exact where the `<script>` tag is re-inserted into the DOM.